orderer/consensus/etcdraft/chain.go

```diff
@@ -921,37 +922,42 @@ func (c *Chain) ordered(msg *orderer.SubmitRequest) (batches [][]*common.Envelop
 			}
 		}
 
-		if !atomic.CompareAndSwapUint32(&c.leaderTransferFailed, 1, 0) {
-			if c.checkForEvictionNCertRotation(msg.Payload) {
-				go func() {
-					if transferee, err := c.Node.abdicateLeader(c.raftID); err != nil {
-						if err.Error() == LEADERTRANSFERTIMEOUT.Error() && transferee != raft.None {
-							oldTransferee := atomic.LoadUint64(&c.leadershipTransferCandidate)
-							if !atomic.CompareAndSwapUint64(&c.leadershipTransferCandidate, oldTransferee, transferee) {
-								atomic.StoreUint32(&c.leaderTransferFailed, 1)
-							}
-							// sleep for tick time & reattempt
-							timer := c.Node.clock.NewTimer(time.Duration(c.Node.config.ElectionTick) * c.Node.tickInterval)
-							defer timer.Stop()
-
-							select {
-							case <-timer.C():
-							case <-c.Node.chain.doneC:
-							}
-						} else if err.Error() == LEADERTRNSFERCALLEDBYFOLLOWER.Error() {
-							c.logger.Debugf("Leadership transfer completed")
-						} else {
-							atomic.StoreUint32(&c.leaderTransferFailed, 1)
-						}
+		if c.checkForEvictionNCertRotation(msg.Payload) && atomic.CompareAndSwapUint32(&c.leadershipTransferInProgress, 0, 1) {
```

Comment:
If another config transaction arrives while the leadershipTransferInProgress, it would proceed further down to propose. 

1) I think, we need to handle it by rejecting it.

Reply:
why?

Reply:
If other transaction is also removing the current leader, it would propose the transaction before transferring the leadership right?